### PR TITLE
file add fix

### PR DESCRIPTION
**Pull request by andrewxhill**

### CODE REVIEW

File: src/core/api.ts [hunk 1]
```diff
@@ -93,7 +93,7 @@ class API {
     }
     return fetch(buildAbsoluteURL(this.baseURL, url), {
       method: 'POST',
-      headers: h,
+      headers: new Headers(h),
```
Comment (andrewxhill) [inline src/core/api.ts:96]: found others having an issue related

File: src/helpers/schema-miller.ts [hunk 2]
```diff
@@ -72,6 +72,10 @@ export default class SchemaMiller {
       }
       if (form && form.getHeaders) {
         headers = form.getHeaders()
+      } else {
+        headers = {
+          'content-type': 'application/json'
+        }
```
Comment (andrewxhill) [inline src/helpers/schema-miller.ts:78]: best place i could find to slip the headers into the mill flow

File: src/modules/files.ts [hunk 3]
```diff
@@ -41,7 +41,7 @@ export default class Files extends API {
    * @returns An array of Thread objects
    */
   async list(thread?: string, offset?: string, limit?: number) {
-    const response = await this.sendGet('api/v0/files', undefined, {
+    const response = await this.sendGet('files', undefined, {
```
Comment (andrewxhill) [inline src/modules/files.ts:44]: forgotten api change i guess

File: src/modules/files.ts [hunk 4]
```diff
@@ -85,7 +85,7 @@ export default class Files extends API {
    * @param caption Caption to associated with the added file object
    * @returns An array of created File objects
    */
-  async addFile(file: any, caption: string, thread?: string) {
+  async addFile(file: any, caption: string, thread?: string): Promise<File> {
```
Comment (andrewxhill) [inline src/modules/files.ts:88]: just moving the response type to the start not end

File: src/modules/files.ts [hunk 5]
```diff
@@ -108,8 +109,8 @@ export default class Files extends API {
       items: [dir]
     }
     const response = await this.sendPost(
-      `api/v0/threads/${thread}/files`, undefined, { caption }, dirs
+      `threads/${thread}/files`, undefined, { caption }, dirs
```
Comment (andrewxhill) [inline src/modules/files.ts:112]: more api changes

**Review by sanderpick — APPROVED** [review]

**Comment by sanderpick** [comment]
Awesome.

**Pull request merged**